core/trino-main/src/main/java/io/trino/operator/scalar/CharacterStringCasts.java

```diff
@@ -111,7 +111,12 @@ public static Slice varcharToCharSaturatedFloorCast(@LiteralParameter("y") long
             return Slices.allocate(toIntExact(y));
         }
 
-        codePoints.set(codePoints.size() - 1, codePoints.get(codePoints.size() - 1) - 1);
+        int lastCodePoint = codePoints.get(codePoints.size() - 1) - 1;
+        if (55296 <= lastCodePoint && lastCodePoint <= 57343) {
```

Comment:
Either use constants if there are (or create them) or use the hex representation with a comment about the significance of these values.

Reply:
ok, I will change it to hex reperesentation

Reply:
Sorry, I'm not really familiar with git, I'm making try to make it right

Reply:
I create a new pull request at https://github.com/trinodb/trino/pull/7734
Please check if it is right this time @hashhar